packages/x-data-grid/src/components/GridHeaders.tsx

```diff
@@ -75,6 +75,7 @@ function GridHeaders() {
       columnVisibility={columnVisibility}
       columnGroupsHeaderStructure={columnGroupsHeaderStructure}
       hasOtherElementInTabSequence={hasOtherElementInTabSequence}
+      {...rootProps.slotProps?.columnHeaders}
```

Comment:
We tend to forget to pass slot props.
Is this a problem we want to solve in a general way?

A few ideas:

1. `Slot` component:
	```tsx
	<Slot
	  slotName="columnHeaders"
	  ref={columnsContainerRef}
	  visibleColumns={visibleColumns}
	  // ...
	/>
	
	function Slot<TSlotName extends keyof GridSlotProps & keyof GridSlotsComponent>({
	  slotName,
	  ...props
	}: GridSlotProps[TSlotName] & { slotName: TSlotName }) {
	  const rootProps = useGridRootProps();
	  const Component = rootProps.slots[slotName] as React.ElementType;
	  return <Component {...props} {...rootProps.slotProps?.[slotName]} />;
	}
	```
2. `useSlot` hook:

	```tsx
	const [Slot, slotProps] = useSlot('columnHeaders');
	// + ESLint rule to enforce the usage of `slotProps`
	
	<Slot
	  slotName="columnHeaders"
	  ref={columnsContainerRef}
	  visibleColumns={visibleColumns}
	  {...slotProps}
	/>
	
	const useSlot = <TSlot extends keyof GridSlotProps & keyof GridSlotsComponent>(slot: TSlot) => {
	  const rootProps = useGridRootProps();
	  const Component = rootProps.slots[slot];
	  const slotProps = rootProps.slotProps?.[slot];
	  return [Component, slotProps] as const;
	};
	```
	
cc @mui/xgrid

Reply:
If we can make an eslint rule for 2, we can probably make one to just enfore it with `<rootProps.slots.x {...rootProps.slotProps.x} />`. I would prefer that. After that option 2. After that option 1.

Reply:
On the pickers we are always using `useSlotProps` (except on one instance of a transition component where it's not really doable).
That way we always have the same DX and we didn't forget to pass any slot props for a very long time.
This is a hook from the core team so it also improve the consistency of the codebase and allow to support callback version of the slotProps (which you don't from what I can see).